atomic_reactor/utils/resource_manager.py

```diff
@@ -121,10 +121,13 @@ def _ssh_run_remote_cmd(self, client, cmd: str):
         """
         logger.debug("Try to run command %s on remote host %s", cmd, self.hostname)
         _, stdout, stderr = client.exec_command(cmd, timeout=SSH_COMMAND_TIMEOUT)
-        return stdout, stderr, stdout.channel.recv_exit_status()
+        exitcode = stdout.channel.recv_exit_status()
+        stdout = stdout.read().decode().strip()
+        stderr = stderr.read().decode().strip()
```

Comment:
Could we first read and then call `recv_exit_status()`? The Paramiko docs have this [warning](https://docs.paramiko.org/en/stable/api/channel.html#paramiko.channel.Channel.recv_exit_status) that recv_exit_status may hang forever unless you receive something from the channel first

Reply:
Good point, I was not aware of that.